### PR TITLE
MAPREDUCE-7281. Fix NoClassDefFoundError on 'mapred minicluster'.

### PR DESCRIPTION
**Pull request by iwasakims**

### CODE REVIEW

**Comment by iwasakims** [comment]
I think I need fix to handle path containing whitespace.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  22m 17s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  13m 54s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| -1 :x: |  shellcheck  |   0m 31s |  The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  shelldocs  |   0m 15s |  There were no new shelldocs issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 46s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 34s |  The patch does not generate ASF License warnings.  |
|  |   |  53m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-2077/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2077 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux d2a976597a13 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c8ed33cd2a4 |
| shellcheck | https://builds.apache.org/job/hadoop-multibranch/job/PR-2077/1/artifact/out/diff-patch-shellcheck.txt |
| Max. process+thread count | 413 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project U: hadoop-mapreduce-project |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-2077/1/console |
| versions | git=2.17.1 maven=3.6.0 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 42s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  17m 37s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  shellcheck  |   0m 39s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  shelldocs  |   0m 15s |  There were no new shelldocs issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  16m 58s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 34s |  The patch does not generate ASF License warnings.  |
|  |   |  38m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-2077/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2077 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 3fba62872cda 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c8ed33cd2a4 |
| Max. process+thread count | 431 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project U: hadoop-mapreduce-project |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-2077/2/console |
| versions | git=2.17.1 maven=3.6.0 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-mapreduce-project/bin/mapred [hunk 1]
```diff
@@ -105,6 +105,8 @@ function mapredcmd_case
     minicluster)
       hadoop_add_classpath "${HADOOP_YARN_HOME}/${YARN_DIR}/timelineservice"'/*'
       hadoop_add_classpath "${HADOOP_YARN_HOME}/${YARN_DIR}/test"'/*'
+      junitjar=$(echo "${HADOOP_TOOLS_LIB_JARS_DIR}"/junit-[0-9]*.jar)
```
Comment (aajisaka) [inline hadoop-mapreduce-project/bin/mapred:108]: I think echo is not required and the following is sufficient:
```
junitjar="${HADOOP_TOOLS_LIB_JARS_DIR}/junit-[0-9]*.jar"
```

testing:
```
bash-5.0$ arg1="foo bar"
bash-5.0$ echo "$arg1"
foo bar
bash-5.0$ arg2="${arg1}/junit-[0-9]*.jar"
bash-5.0$ echo "$arg2"
foo bar/junit-[0-9]*.jar
```

`${junitjar}` must be double-quoted.

Comment (aajisaka) [reply hadoop-mapreduce-project/bin/mapred:108]: Sorry, echo is required to resolve the jar file.

Comment (iwasakims) [reply hadoop-mapreduce-project/bin/mapred:108]: We need shell filename expansion before adding the path to CLASSPATH. Java only accepts `/path/to/*` wildcard.

**Comment by iwasakims** [comment]
Thanks, @aajisaka.

**Pull request merged**